app/uk/gov/hmrc/teamsandrepositories/connectors/GithubConnector.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.teamsandrepositories.connectors
+
+import javax.inject.Inject
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.logging.Authorization
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, NotFoundException}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.teamsandrepositories.config.GithubConfig
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GithubConnector @Inject()(githubConfig: GithubConfig, http: HttpClient)(implicit ec: ExecutionContext) {
+
+  def getFileContent(repoName: String, path: String): Future[Option[String]] = {
+    implicit val hc = HeaderCarrier().withExtraHeaders("Authorization" -> s"token ${githubConfig.githubApiOpenConfig.key}")
```

Comment:
I might be mis-remembering this but does http verbs do something weird when setting the Authorization header? I seem to remember there being some issues with the artifactory connector back in the day?